client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -63,17 +64,17 @@ const MediaModalImageEditorCanvas = React.createClass( {
 	},
 
 	componentWillReceiveProps( newProps ) {
-		if ( this.props.src !== newProps.src ) {
+		if ( this.state.src !== newProps.src ) {
 			this.getImage( newProps.src );
 		}
 	},
 
 	componentDidMount() {
-		if ( ! this.props.src ) {
+		if ( ! this.state.src || ( this.state.src === this.props.src ) ) {
```

Comment:
Yeah, great catch. The problem is that when you click on "Edit Image" after previously editing some image, the image `src` is initially pointing to the old image. However, right after the image editor is mounted, the `src` prop is sent second time with the new/correct image to edit.

I fixed it in b6079b6 by loading and rendering the image only in `componentWillReceiveProps` which gets called only the second time props are sent (with the correct image URL).

But I wonder, maybe I should dig deeper and look for the root cause of why we first send the old image and the new/correct image only in the second time.